app/javascript/application.js

```diff
@@ -3,3 +3,14 @@ import "@hotwired/turbo-rails"
 import "controllers"
 import "@popperjs/core"
 import "bootstrap"
+
+import $ from 'jquery';
+import 'select2';
+
+$(document).on('turbo:load', function() {
+  $('.select2').select2({
+    width: '100%',
+    placeholder: "Search for a book...",
+    allowClear: true
+  });
+});
```

Comment:
@liamgingell 

Instead of `jquery` when we want to use javascript. Please use stimulus 🙏